src/components/PopupWithForm.js

```diff
@@ -1,39 +1,48 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(modalSelector, handleFormSubmit) {
-    super(modalSelector);
-    this._modalForm = this._modal.querySelector(".modal__form");
-    this._inputList = [...this._modalForm.querySelectorAll(".modal__input")];
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._modalForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._modalForm.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
     this._handleFormSubmit = handleFormSubmit;
   }
-
-  close() {
-    super.close(this._modal);
-  }
-
-  setLoading(isLoading, text) {
-    const modalButton = this._modal.querySelector(".modal__button");
-    console.log(modalButton);
-    modalButton.textContent = isLoading ? text : "Save";
-  }
-
   _getInputValues() {
-    const formValues = {};
-    this._inputList.forEach((inputEl) => {
-      formValues[inputEl.name] = inputEl.value;
+    const inputValues = {};
+    this._modalForm.querySelectorAll(".modal__input").forEach((input) => {
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.